mantis-server/mantis-server-worker-client/src/main/java/io/mantisrx/server/worker/client/MantisHttpClientImpl.java

```diff
@@ -0,0 +1,143 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.mantisrx.server.worker.client;
+
+import io.netty.bootstrap.Bootstrap;
+import io.netty.channel.Channel;
+import io.netty.channel.ChannelInitializer;
+import java.util.ArrayList;
+import java.util.List;
+import mantis.io.reactivex.netty.channel.ObservableConnection;
+import mantis.io.reactivex.netty.client.ClientChannelFactory;
+import mantis.io.reactivex.netty.client.ClientConnectionFactory;
+import mantis.io.reactivex.netty.client.ClientMetricsEvent;
+import mantis.io.reactivex.netty.client.ConnectionPool;
+import mantis.io.reactivex.netty.client.ConnectionPoolBuilder;
+import mantis.io.reactivex.netty.metrics.MetricEventsSubject;
+import mantis.io.reactivex.netty.pipeline.PipelineConfigurator;
+import mantis.io.reactivex.netty.protocol.http.client.HttpClientImpl;
+import mantis.io.reactivex.netty.protocol.http.client.HttpClientRequest;
+import mantis.io.reactivex.netty.protocol.http.client.HttpClientResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import rx.Observable;
+
+public class MantisHttpClientImpl<I, O> extends HttpClientImpl<I, O> {
+    private static final Logger logger = LoggerFactory.getLogger(MantisHttpClientImpl.class);
+    protected final String name;
+    protected final ServerInfo serverInfo;
+    protected final Bootstrap clientBootstrap;
+    protected final PipelineConfigurator<HttpClientResponse<O>, HttpClientRequest<I>> pipelineConfigurator;
+    protected final ClientChannelFactory<HttpClientResponse<O>, HttpClientRequest<I>> channelFactory;
+    protected final ClientConnectionFactory<HttpClientResponse<O>, HttpClientRequest<I>, ? extends ObservableConnection<HttpClientResponse<O>, HttpClientRequest<I>>> connectionFactory;
+    protected final ClientConfig clientConfig;
+    protected final MetricEventsSubject<ClientMetricsEvent<?>> eventsSubject;
+    protected final ConnectionPool<HttpClientResponse<O>, HttpClientRequest<I>> pool;
+
+    private Observable<ObservableConnection<HttpClientResponse<O>, HttpClientRequest<I>>> observableConection;
+    private List<Channel> connectionTracker;
+
+    public MantisHttpClientImpl(String name, ServerInfo serverInfo, Bootstrap clientBootstrap, PipelineConfigurator<HttpClientResponse<O>, HttpClientRequest<I>> pipelineConfigurator, ClientConfig clientConfig, ClientChannelFactory<HttpClientResponse<O>, HttpClientRequest<I>> channelFactory, ClientConnectionFactory<HttpClientResponse<O>, HttpClientRequest<I>, ? extends ObservableConnection<HttpClientResponse<O>, HttpClientRequest<I>>> connectionFactory, MetricEventsSubject<ClientMetricsEvent<?>> eventsSubject) {
+        super(name, serverInfo, clientBootstrap, pipelineConfigurator, clientConfig, channelFactory, connectionFactory, eventsSubject);
+
+        this.connectionTracker = new ArrayList<>();
+
+        if (null == name) {
```

Comment:
I don't think this whole section is needed. Most of these instance variables are not used anywhere inside this class. The `super` call above should've handled all initializations for you.